substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Resources.java

```diff
@@ -71,7 +74,12 @@ public static Resources singleton() {
      * resourceName) provides implementations for {@code hashCode()} and {@code equals()} needed for
      * the map keys.
      */
-    private final EconomicMap<Pair<String, String>, ResourceStorageEntry> resources = ImageHeapMap.create();
+    private final EconomicMap<Pair<String, String>, Object> resources = ImageHeapMap.create();
+    private final List<Pair<String, Pattern>> includePatterns = new ArrayList<>();
+    private final List<Pair<String, Pattern>> excludePatterns = new ArrayList<>();
+
+    public static final Object NEGATIVE_QUERY = new Object();
```

Comment:
A comment here explaining where `NEGATIVE_QUERY` and `MISSING_METADATA` are used and why they are never expected in the same place would be useful.